src/librustdoc/json/mod.rs

```diff
@@ -170,12 +170,9 @@ impl<'tcx> FormatRenderer<'tcx> for JsonRenderer<'tcx> {
             } else if let types::ItemEnum::Enum(ref mut e) = new_item.inner {
                 e.impls = self.get_impls(id)
             }
-            let removed = self.index.borrow_mut().insert(from_def_id(id), new_item.clone());
-            // FIXME(adotinthevoid): Currently, the index is duplicated. This is a sanity check
-            // to make sure the items are unique.
-            if let Some(old_item) = removed {
-                assert_eq!(old_item, new_item);
-            }
+            let removed = self.index.borrow_mut().insert(from_def_id(id), new_item);
+
+            assert_matches!(removed, None);
```

Comment:
```suggestion

            assert!(removed.is_none());
```

Reply:
This gives a nicer panic messages when `removed` is `Some`